rfd/xxxx-tsh-bench-db.md

```diff
@@ -0,0 +1,249 @@
+    tsh->>Teleport: Establish connection
+    Teleport->>DB: Connect
+    DB->>tsh: Connection
+    tsh->>DB: Ping
+  end
+```
+
+#### Import/Export data
+It is a typical flow for users to import data to their databases. This suite is
+focused on running multiple queries within the same database connection (much
+like what an import tool would do). To do so, some databases will require an
+additional step executed before the suite starts. For example, on PostgreSQL,
+the necessary tables must be created. This setup won't be performed inside the
+workload function.
+
+The exporting suite will load data outside the workload function. This way, the
+measuring will only include read queries.
+
+```code
+$ tsh bench db import --db-user=postgres --db-name=postgres postgres-dev
+```
+
+```mermaid
+sequenceDiagram
+  participant tsh
+  participant Teleport
+  participant Database as DB
+
+  tsh->>Teleport: Generate Certificates (tsh db login)
+  tsh->>Teleport: Establish connection
+  Teleport->>DB: Connect
+  DB->>tsh: Connection
+
+  loop every execution
+    tsh->>DB: Query
+    DB->>tsh: Result
+  end
+```
+
+##### Data generation
+The database clients will have access to a set of functions that can be used to
+generate random data. Those functions will cover the following types: `[]byte`,
+`string`,  `bool`, `int64`, and `float64`. Some of those functions, such as
+generating `[]byte` and `string`, will make it possible to define their length.
+The clients will be responsible for determining their structure (if required,
+creating them on the `Setup` call) and filling with data on `CreateRecord`.
+
+Users will be able to seed them through the `--seed` flag. If none is provided,
+a random seed will be used. In addition, the seed will be printed so it can be
+used to replicate failures.
+
+### Direct database connections
+To directly connect to the databases, the suites must accept the credentials
+that will be used. To avoid adding multiple flags to cover the scenarios, we're
+heading into having a single `--uri` flag, which will receive the database URI
+with all options included. This way, users can choose the authentication method
+that will be used.
+
+```code
+$ tsh bench db connect --uri postgres://hello@localhost:5432
+```
+
+By providing this flag, the `tsh` will connect directly to the database without
+using any Teleport certificate.
+
+## Security
+
+### Bypass MFA or access checker
+The command will use the same flow used by `tsh db login` to issue certificates,
+including support to MFA. The main difference is that the issue credentials
+won't be persisted on disk.
+
+### Malicious URI on the direct database connection
+Users may try to send malicious URI when connecting directly to databases.
+Teleport doesn't validate the input provided; it is passed directly to the
+database client responsible for validating it. Client execution is entirely
+done locally. Nothing is sent to the Teleport server.
+
+## Future work
```

Comment:
Thinking about this further, it would be useful to test how well we scale with the number of clients. We can add `--scale=N` param and execute `N` tests in parallel. This sort of testing would highlight bottlenecks to be optimized.